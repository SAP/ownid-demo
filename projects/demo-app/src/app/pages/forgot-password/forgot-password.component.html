<demo-header></demo-header>

<div class="container">
  <div class="content">
    <form class="own-form" [formGroup]="form" (ngSubmit)="onSubmit()">
      <ng-container *ngIf="!emailSent; else emailWasSent">
        <div class="title">Canâ€™t login?</div>

        <div class="description">Enter in your email to reset access to your account.</div>

        <div class="errors" *ngIf="errors">{{errors}}</div>

        <input class="own-input" type="email" placeholder="Email" formControlName="email">

        <button class="own-button" type="submit">Continue</button>
      </ng-container>
    </form>
  </div>

  <ng-template #emailWasSent>
    <div class="title">Email Sent</div>

    <div class="description send">
      Your recovery link has been sent to
      <br>
      <strong>{{form.value.email}}</strong>.
    </div>

    <svg xmlns="http://www.w3.org/2000/svg" width="136" height="120" class="send-icon">
      <path
          d="M40.174 32.046c-3.287 0-6.087 2.8-6.087 6.087V81.96c0 3.287 2.8 6.087 6.087 6.087h65.74c3.287 0 6.087-2.8 6.087-6.087V38.133c0-3.287-2.8-6.087-6.087-6.087h-65.74zm0 4.87h65.74c.73 0 1.217.487 1.217 1.217V81.96c0 .73-.487 1.217-1.217 1.217h-65.74c-.73 0-1.217-.487-1.217-1.217V38.133c0-.73.487-1.217 1.217-1.217zM49.67 44.1c-1.34.122-2.313 1.34-2.19 2.678 0 .61.365 1.096.73 1.583l12.296 12.174L48.33 71.612c-.974.852-1.096 2.435-.243 3.41s2.435 1.096 3.41.243l12.417-11.2 3.652 3.652c3.043 2.922 7.913 2.922 10.835 0l3.652-3.652 12.417 11.2c.974.852 2.557.852 3.41-.122s.852-2.556-.122-3.41L85.583 60.655 97.878 48.48c.974-.974.974-2.435.122-3.41-.974-.974-2.435-.974-3.41-.122l-.122.122L74.99 64.307c-.852.852-3.043.852-3.896 0L51.74 45.072a2.57 2.57 0 0 0-2.07-.974z"
          fill="#0070f2"/>
      <path fill="#64edd2"
            d="M0 45.437c0-1.34 1.096-2.435 2.435-2.435h24.348c1.34 0 2.435 1.096 2.435 2.435s-1.096 2.435-2.435 2.435H2.435C1.096 47.872 0 46.776 0 45.437zm4.87 14.61c0-1.34 1.096-2.435 2.435-2.435h19.478c1.34 0 2.435 1.096 2.435 2.435s-1.096 2.435-2.435 2.435H7.304c-1.34 0-2.435-1.096-2.435-2.435zm4.87 14.61c0-1.34 1.096-2.435 2.435-2.435h14.6c1.34 0 2.435 1.096 2.435 2.435S28.124 77.1 26.785 77.1h-14.6c-1.34 0-2.435-1.096-2.435-2.435z"/>
    </svg>

    <div class="row">
      <button class="own-button" type="button" (click)="onSubmit()">Resend Email</button>

      <button class="own-flat-button" type="button" (click)="onGoToLogin()">Go to login page</button>
    </div>
  </ng-template>
</div>
<demo-footer></demo-footer>


