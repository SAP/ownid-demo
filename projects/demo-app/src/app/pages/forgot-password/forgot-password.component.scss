:host {
  background: #E5E5E5;
}

.logo {
  margin: 0 auto;
}

.title {
  font-size: 20px;
  line-height: 32px;
  font-weight: bold;
}

.description {
  font-size: 14px;
  line-height: 20px;
  margin-bottom: 32px;

  &.send {
    margin-bottom: 16px;
  }
}

.container {
  max-width: 480px;
  width: calc(100% - 16px);
  margin: 48px auto 0;
}

.content {
  color: #111D29;
  margin-top: 40px;
  padding: 24px 24px 32px;

  background: #FFFFFF;
  box-shadow: 0 8px 16px rgba(4, 172, 167, 0.16), 0 0 2px rgba(4, 172, 167, 0.16);
  border-radius: 6px;
}

.send-icon {
  width: 136px;
  margin: 0 auto;
}