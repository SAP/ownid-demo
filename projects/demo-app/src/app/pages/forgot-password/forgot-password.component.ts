import { Component } from '@angular/core';
import { FormBuilder, FormGroup, Validators } from '@angular/forms';
import { Router } from '@angular/router';
import { GigyaService } from '../../services/gigya.service';

@Component({
  selector: 'login',
  templateUrl: './forgot-password.component.html',
  styleUrls: ['./forgot-password.component.scss']
})
export class ForgotPasswordComponent {
  form: FormGroup;

  emailSent = false;

  errors: string | null = null;

  constructor(
    formBuilder: FormBuilder,
    private gigyaService: GigyaService,
    private router : Router,
  ) {
    this.form = formBuilder.group({
      email: ['', [Validators.email, Validators.required]],
    });
  }

  onSubmit() {
    if (this.form.valid) {
      this.errors = null;

      const params = {
        loginID: this.form.value.email,
      }

      // eslint-disable-next-line @typescript-eslint/no-explicit-any
      this.gigyaService.resetPassword(params, (data: any) => {
        if (data.status === 'FAIL') {
          this.errors = data.errorDetails;
          return;
        }

        this.emailSent = true;
      });
    }
  }

  onGoToLogin() {
    this.router.navigateByUrl('/')
  }
}
