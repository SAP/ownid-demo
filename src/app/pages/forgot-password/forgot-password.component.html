<section class="sign-body">
  <div class="container">
    <div class="call-2-action">
      <h1 class="headline">
        For the best experience use Own Id for login
      </h1>
      <h2 class="sub-title">
        Scan QR code with your phone and proceed with out any logins and
        passwords
      </h2>
    </div>
    <div>
      <mat-card class="sign-section">
        <mat-card-title>
          Recover account
        </mat-card-title>

        <ng-container *ngIf="!emailSent; else emailWasSent">
          <form [formGroup]="form" (ngSubmit)="onSubmit()">
            <p>
              <span class="errors" *ngIf="errors">{{errors}}</span>
            </p>
            <p>
              <mat-form-field class="form-field">
                <mat-label>Enter your email</mat-label>
                <input
                    matInput
                    placeholder="Email"
                    formControlName="email"
                    required
                />
                <mat-error *ngIf="form.controls.email.hasError('required')">You must enter a value</mat-error>
                <mat-error *ngIf="form.controls.email.hasError('email')">Not a valid email</mat-error>
              </mat-form-field>
            </p>
            <button mat-flat-button color="primary" class="form-field" type="submit">Reset Password</button>
          </form>
        </ng-container>

        <ng-template #emailWasSent>
          <div class="text">Recover account link was sent to you email address.</div>
        </ng-template>
      </mat-card>
    </div>
  </div>
</section>



