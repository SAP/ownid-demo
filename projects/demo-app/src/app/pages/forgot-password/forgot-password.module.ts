import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { ReactiveFormsModule } from '@angular/forms';
import { OwnidModule } from '../../shared/ownid/ownid.module';
import { ForgotPasswordRoutingModule } from './forgot-password-routing.module';
import { ForgotPasswordComponent } from './forgot-password.component';

@NgModule({
  declarations: [ForgotPasswordComponent],
  imports: [
    CommonModule,
    ForgotPasswordRoutingModule,
    ReactiveFormsModule,
    OwnidModule,
  ]
})
export class ForgotPasswordModule { }
