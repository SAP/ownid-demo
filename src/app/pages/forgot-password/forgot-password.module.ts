import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { ReactiveFormsModule } from '@angular/forms';
import { MatCardModule } from '@angular/material/card';
import { MatInputModule } from '@angular/material/input';
import { MatButtonModule } from '@angular/material/button';

import { ForgotPasswordRoutingModule } from './forgot-password-routing.module';
import { ForgotPasswordComponent } from './forgot-password.component';


@NgModule({
  declarations: [ForgotPasswordComponent],
  imports: [
    CommonModule,
    ForgotPasswordRoutingModule,
    MatCardModule,
    MatInputModule,
    ReactiveFormsModule,
    MatButtonModule,
  ]
})
export class ForgotPasswordModule { }
