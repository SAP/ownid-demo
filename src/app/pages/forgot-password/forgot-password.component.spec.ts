import { async, TestBed } from "@angular/core/testing";
import { NO_ERRORS_SCHEMA } from "@angular/core";
import { FormBuilder } from "@angular/forms";

import { GigyaService } from "../../services/gigya.service";
import { ForgotPasswordComponent } from "./forgot-password.component";

describe("ForgotPasswordComponent", () => {
  let fb: FormBuilder;
  let gigyaService: GigyaService;

  beforeEach(() => {
    fb = {} as FormBuilder;
    fb.group = jest.fn().mockReturnValue({ valid: true, value: {} });

    gigyaService = {} as GigyaService;
    gigyaService.resetPassword = jest
      .fn()
      .mockImplementation(
        (params: never, callBack: (data: unknown) => void) => {
          callBack({ status: "OK" });
        }
      );
  });

  describe("Snapshot", () => {
    beforeEach(async(() => {
      TestBed.configureTestingModule({
        imports: [],
        providers: [
          {
            provide: FormBuilder,
            useValue: fb
          },
          {
            provide: GigyaService,
            useValue: gigyaService
          }
        ],
        declarations: [ForgotPasswordComponent],
        schemas: [NO_ERRORS_SCHEMA]
      }).compileComponents();
    }));

    test("check snapshot", () => {
      const fixture = TestBed.createComponent(ForgotPasswordComponent);

      expect(fixture).toMatchSnapshot();
    });
  });

  describe("controller", () => {
    it("should set form", () => {
      const sut = new ForgotPasswordComponent(fb, gigyaService);

      expect(sut.form).toBeTruthy();
    });
  });

  describe("onSubmit", () => {
    it("should set emailSent to true", () => {
      const sut = new ForgotPasswordComponent(fb, gigyaService);

      sut.onSubmit();

      expect(sut.emailSent).toEqual(true);
    });

    it("should set errors on gigya error", () => {
      const sut = new ForgotPasswordComponent(fb, gigyaService);

      gigyaService.resetPassword = jest
        .fn()
        .mockImplementation(
          (params: never, callBack: (data: unknown) => void) => {
            callBack({ status: "FAIL", errorDetails: { errors: [true] } });
          }
        );

      sut.onSubmit();

      expect(sut.errors).toEqual({ errors: [true] });
      expect(sut.emailSent).toEqual(false);
    });

    it("should not update emailSent when form is invalid", () => {
      fb.group = jest.fn().mockReturnValue({ valid: false });

      const sut = new ForgotPasswordComponent(fb, gigyaService);

      sut.onSubmit();

      expect(sut.emailSent).toEqual(false);
    });
  });
});
