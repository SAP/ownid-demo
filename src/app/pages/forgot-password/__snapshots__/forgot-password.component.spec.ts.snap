// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ForgotPasswordComponent Snapshot check snapshot 1`] = `
<login
  emailSent="false"
  errors="null"
  form={[Function Object]}
  gigyaService={[Function Object]}
>
  <section
    class="sign-body"
  >
    <div
      class="container"
    >
      <div
        class="call-2-action"
      >
        <h1
          class="headline"
        >
           For the best experience use Own Id for login 
        </h1>
        <h2
          class="sub-title"
        >
           Scan QR code with your phone and proceed with out any logins and passwords 
        </h2>
      </div>
      <div>
        <mat-card
          class="sign-section"
        >
          <mat-card-title>
             Recover account 
          </mat-card-title>
          
          
        </mat-card>
      </div>
    </div>
  </section>
</login>
`;
